### PR TITLE
- Added NaN cleaning deep learning models, retaining 66.5% of NaN dat…

### PR DESCRIPTION
**Pull request by hashemJaber**
…aset for 30 most issue-causing columns

- Added new dataset with NaN values replaced with deep learning predicted values for main model training
- Added time processing tools for positive and negative scenarios
- Added automated model training for linear regression tasks for acres burned prediction
- Added automated model training for logistic regression tasks for fire prediction
- Added evaluation scripts
- Trained Models
- Added ensemble voting logic combining deep learning and SGDClassifier models  #TODO ADD MORE MODELS
- Added conversion of model outputs to class labels for evaluation compatibility
- Added training and evaluation procedures for train, validation, and test sets
- Added model evaluation metrics: accuracy, precision, recall, F1 score, and confusion matrix
- Ensured compatibility between PyTorch and sklearn model outputs
- Integrated model predictions into evaluation workflow using thresholded ensemble output

### CODE REVIEW

**Comment by Shawn-Kim96** [comment]
@hashemJaber Great work, just a few things to point out
1. Since we are going to have various data processing scripts, can you add functions as scripts under `src/dataset_preprocessing/` in python files?
2. Can you save models under `models/...` so that we can load model whenever we want?
